elasticsearch/Dockerfile

```diff
@@ -119,24 +71,40 @@ RUN sed -i -e 's/ES_DISTRIBUTION_TYPE=tar/ES_DISTRIBUTION_TYPE=docker/' bin/elas
     find config -type f -exec chmod 0664 {} +
 
 ################################################################################
-# Stage 3. Build the final image, using the rootfs above as the basis, and
-# copying in the Elasticsearch distribution
+# Build stage 1 (the actual Elasticsearch image):
+#
+# Copy elasticsearch from stage 0
+# Add entrypoint
 ################################################################################
-FROM scratch
 
-# Setup the initial filesystem.
-COPY --from=rootfs /rootfs /
+FROM ubuntu:20.04
+
+RUN for iter in 1 2 3 4 5 6 7 8 9 10; do \
+      export DEBIAN_FRONTEND=noninteractive && \
+apt-get update && \
+apt-get upgrade -y && \
+apt-get install -y --no-install-recommends curl netcat zip unzip vim-tiny  && \
+apt-get clean && \
+rm -rf /var/lib/apt/lists/* && \
+      exit_code=0 && break || \
+        exit_code=$? && echo "apt-get error: retry $iter in 10s" && sleep 10; \
+    done; \
+    exit $exit_code
 
 RUN groupadd -g 1000 elasticsearch && \
-    adduser -u 1000 -g 1000 -G 0 -d /usr/share/elasticsearch elasticsearch && \
-    chmod 0755 /usr/share/elasticsearch && \
+    adduser --uid 1000 --gid 1000 --home /usr/share/elasticsearch elasticsearch && \
+    adduser elasticsearch root && \
     chown -R 0:0 /usr/share/elasticsearch
 
 ENV ELASTIC_CONTAINER true
 
 WORKDIR /usr/share/elasticsearch
 COPY --from=builder --chown=0:0 /usr/share/elasticsearch /usr/share/elasticsearch
 
+COPY --from=builder --chown=0:0 /bin/tini /bin/tini
+
+COPY --from=builder --chown=0:0 /etc/ssl/certs/java/cacerts /etc/ssl/certs/java/cacerts
```

Comment:
This is really curious -- can you elaborate on why this was added in this way instead of installing something like `p11-kit` to use the `trust` binary to keep this file automatically in sync with the "system" `ca-certificates` store?  That would allow users to install new certificates in the usual way and only require that they run `update-ca-certificates` afterwards (which is a pretty standard workflow).

See https://github.com/docker-library/openjdk/blob/62b5c6a9982bcb9d0c9a1503a3abb1b7cef6ca35/11/jre/bullseye/Dockerfile#L88-L99 for an example of how that works.

Reply:
Ah, thanks @tianon - there some, uh, history about that particular line, but your suggestion is much better. We've already signed off 7.16.1, so I hope you don't mind if we include that from the next release?

Reply:
I answered in another thread, but I'll mirror here for continuity that in light of the log4j stuff, I agree. :smile: